dit/distconst.py

```diff
@@ -982,3 +982,30 @@ def random_dist_structure(outcome_length, alphabet_size):
     """
     bound = 2**(alphabet_size**outcome_length)
     return _int_to_dist(randint(0, bound), outcome_length, alphabet_size)
+
+def _combine_scalar_dists(d1, d2, op):
+    """
+    Combines `d1` and `d2` according to `op`, as though they are independent.
+
+    Parameters
+    ----------
+    d1 : ScalarDistribution
+        The first distribution
+    d2 : ScalarDistribution
+        The second distribution
+    op : function
+        Function used to combine outcomes
+
+    Returns
+    -------
+    d : ScalarDistribution
+        The two distributions combined via `op`
+    """
+    # Copy to make sure we don't lose precision when converting.
+    d2 = d2.copy(base=d1.get_base())
+
+    dist = defaultdict(float)
+    for (o1, p1), (o2, p2) in product(d1.zipped(), d2.zipped()):
+        dist[op(o1, o2)] += d1.ops.mult(p1, p2)
```

Comment:
I was just about to comment that this needs to handle logprobs, but then I remembered...we solved that problem already. Haha.